pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```
@@ -1097,17 +1097,16 @@ private void createHelixEntriesForHighLevelConsumer(AbstractTableConfig config,
     }
   }
 
-  public void setTableConfig(AbstractTableConfig config, String tableNameWithSuffix, TableType type)
-      throws JsonGenerationException, JsonMappingException, IOException {
+  public void setExistingTableConfig(AbstractTableConfig config, String tableNameWithSuffix, TableType type)
+      throws IOException {
     if (type == TableType.REALTIME) {
       ZKMetadataProvider.setRealtimeTableConfig(_propertyStore, tableNameWithSuffix,
           AbstractTableConfig.toZnRecord(config));
       ensureRealtimeClusterIsSetUp(config, tableNameWithSuffix, config.getIndexingConfig());
-    } else {
+    } else if (type == TableType.OFFLINE) {
       ZKMetadataProvider.setOfflineTableConfig(_propertyStore, tableNameWithSuffix,
           AbstractTableConfig.toZnRecord(config));
     }
```

Should throw an exception if it's neither realtime/offline